plugins/runtime_id_plugin/runtime_id_plugin.c

```diff
@@ -0,0 +1,36 @@
+#include "runtime_id_plugin.h"
+
+#include <sodium.h>
+
+#include "uuid/uuid.h"
+
+datadog_php_uuid runtime_id = DATADOG_PHP_UUID_INIT;
+
+typedef enum datadog_php_plugin_result plugin_result_t;
+
+static plugin_result_t datadog_php_runtime_id_plugin_minit(int type, int module_number) {
+    (void)type;
+    (void)module_number;
+
+    if (sodium_init() == -1) {
+        return DATADOG_PHP_PLUGIN_FAILURE;
+    }
+
+    _Alignas(16) uint8_t bytes[16];
+    randombytes_buf(bytes, sizeof bytes);
+
+    datadog_php_uuidv4_bytes_ctor(&runtime_id, bytes);
+
+    return DATADOG_PHP_PLUGIN_SUCCESS;
+}
+
+static void datadog_php_runtime_id_plugin_mshutdown(int type, int module_number) {
+    (void)type;
+    (void)module_number;
+    randombytes_close();
```

Comment:
Do we need to wait until MSHUTDOWN to call this or can we call it right after the `randombytes_buf()` call? From [the docs](https://libsodium.gitbook.io/doc/generating_random_data) it seems like we might not need to call it at all?

> This deallocates the global resources used by the pseudo-random number generator. More specifically, when the `/dev/urandom` device is used, it closes the descriptor. Explicitly calling this function is almost never required.

Reply:
My worry with closing it earlier would possibly be closing `/dev/urandom` prematurely for PHP's sodium ext. Since init and close are global (no resource handle directly exposed to users), I could theoretically mess it up for others.

I have not tested this, though; just a theory.

Reply:
Ah, good point. Do you think it is safe to call it at all then? I'm trying to think of a scenario where another extension uses sodium in their MSHUTDOWN after ours fires or something.

Reply:
This is exactly why global shared state is bad. The docs do not say if it's reference counted or anything, so I don't know the answer! We'll have to look at the code.